### PR TITLE
Issues/#129 improve swagger document generation

### PR DESCRIPTION
**Pull request by ghost**
**Demo:**
image
image

**Metadata api call:**
image

**Api call:**
image

**Failed generation case:**
image

**Failed generation case with ILogger:**
image

### CODE REVIEW

File: src/Nox.Api.OData/Swagger/ODataCustomSwaggerFilter.cs [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            RoutingConstants.PropertyParameterName,
+            RoutingConstants.NavigationParameterName
+        };
+
+        public void Apply(OpenApiDocument swaggerDoc, DocumentFilterContext context)
+        {
+            if (_noxConfiguration?.Entities == null ||
+                !_noxConfiguration.Entities.Any())
+            {
+                Console.ForegroundColor = ConsoleColor.Yellow;
```
Comment (andresharpe) [inline src/Nox.Api.OData/Swagger/ODataCustomSwaggerFilter.cs]: Should we not inject and output to ILogger rather than Console?

**Comment by andresharpe** [comment]
We need to remove the null dereferencing warning and just resolve merge conflicts. See my ILogger rather than Console.Writeline comeent. Rest looks good thanks Dmytro

**Comment by ghost** [comment]
@andresharpe This seems to be resolved now. I can't trigger the pipeline to find it out since I'm not a repo maintainer. Also, I can't see your comment, so I just changed the code a bit to have the option to utilize ILogger.

Please, add the team and me as repo maintainers.

Thanks!

**Review by andresharpe — COMMENTED** [review]
Looks good

**Pull request merged**